From 8f2053e60d8412024b6097368748ecaf5d329599 Mon Sep 17 00:00:00 2001
Message-ID: <8f2053e60d8412024b6097368748ecaf5d329599.1762612978.git.lorenzo@kernel.org>
From: Lorenzo Bianconi <lorenzo@kernel.org>
Date: Fri, 24 Oct 2025 17:39:36 +0200
Subject: [PATCH 01/12] mt76: dma: always enable RRO queues for non-MT7992
 chipset

Signed-off-by: Lorenzo Bianconi <lorenzo@kernel.org>
---
 dma.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/dma.c b/dma.c
index f240016e..ba84f619 100644
--- a/dma.c
+++ b/dma.c
@@ -6,6 +6,7 @@
 #include <linux/dma-mapping.h>
 #include "mt76.h"
 #include "dma.h"
+#include "mt76_connac.h"
 
 static struct mt76_txwi_cache *
 mt76_alloc_txwi(struct mt76_dev *dev)
@@ -189,7 +190,8 @@ static void
 mt76_dma_sync_idx(struct mt76_dev *dev, struct mt76_queue *q)
 {
 	Q_WRITE(q, desc_base, q->desc_dma);
-	if ((q->flags & MT_QFLAG_WED_RRO_EN) && !mt76_npu_device_active(dev))
+	if ((q->flags & MT_QFLAG_WED_RRO_EN) &&
+	    (!is_mt7992(dev) || !mt76_npu_device_active(dev)))
 		Q_WRITE(q, ring_size, MT_DMA_RRO_EN | q->ndesc);
 	else
 		Q_WRITE(q, ring_size, q->ndesc);
-- 
2.51.1

