diff --git a/mt7996/mac.c b/mt7996/mac.c
index 753c6987..422f1459 100644
--- a/mt7996/mac.c
+++ b/mt7996/mac.c
@@ -11,6 +11,9 @@
 #include "mac.h"
 #include "mcu.h"
 
+static bool debug;
+module_param(debug, bool, 0644);
+
 #define to_rssi(field, rcpi)	((FIELD_GET(field, rcpi) - 220) / 2)
 
 static const struct mt7996_dfs_radar_spec etsi_radar_specs = {
@@ -1637,6 +1640,7 @@ bool mt7996_rx_check(struct mt76_dev *mdev, void *data, int len)
 
 	switch (type) {
 	case PKT_TYPE_TXRX_NOTIFY:
+		if (debug) printk("[%s-%d] PKT_TYPE_TXRX_NOTIFY\n", __func__, __LINE__);
 		mt7996_mac_tx_free(dev, data, len);
 		return false;
 	case PKT_TYPE_TXS:
@@ -1670,6 +1674,7 @@ void mt7996_queue_rx_skb(struct mt76_dev *mdev, enum mt76_rxq_id q,
 
 	switch (type) {
 	case PKT_TYPE_TXRX_NOTIFY:
+		if (debug) printk("[%s-%d] PKT_TYPE_TXRX_NOTIFY\n", __func__, __LINE__);
 		if (mtk_wed_device_active(&dev->mt76.mmio.wed_hif2) &&
 		    q == MT_RXQ_TXFREE_BAND2) {
 			dev_kfree_skb(skb);
